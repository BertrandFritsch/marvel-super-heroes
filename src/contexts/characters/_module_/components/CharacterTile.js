// @flow

import React from 'react';
import type { Item } from '../flowTypes';

type Props = {

  /**
   * The item
   */
  item: Item,

  /**
   * Called when the component has been loaded
   */
  onTileDetailsRequested: (item: Item) => void
};

export default function CharacterTile(props: Props) {
  const { item } = props;

  return (
    <div className="character-tile">
      <a href="#" onClick={ () => props.onTileDetailsRequested(item) }>
        <img src={ item.thumbnail } />
      </a>
      <span className="character-tile-name">{item.name}</span>
      <div className="character-tile-links">
        { item.urls.map(link => (<a key={ link.type } href={ link.url }>{ link.type }</a>)) }
      </div>
    </div>
  );
}
