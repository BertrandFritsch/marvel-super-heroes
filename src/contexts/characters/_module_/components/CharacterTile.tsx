import * as React from "react";
import { Character } from "../reducers";

interface Props {

  /**
   * The item
   */
  item: Character;

  /**
   * Called when the component has been loaded
   */
  onTileDetailsRequested: (item: Character) => void;
}

export default function CharacterTile(props: Props) {
  const { item } = props;

  return (
    <div className="character-tile">
      <a href="#" onClick={ () => props.onTileDetailsRequested(item) }>
        <img src={ item.thumbnail } />
      </a>
      <span className="character-tile-name">{item.name}</span>
      <div className="character-tile-links">
        { item.urls.map((link) => (<a key={ link.type } href={ link.url }>{ link.type }</a>)) }
      </div>
    </div>
  );
}
