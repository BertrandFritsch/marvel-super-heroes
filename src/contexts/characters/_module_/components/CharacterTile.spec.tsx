import { mount } from 'enzyme';
import * as React from 'react';
import CharacterTile from './CharacterTile';

describe('CharacterTile', () => {
  const character = {
    id: 'id1',
    name: 'name1',
    description: 'desc 1',
    thumbnail: 'http://.../standard_xlarge.jpg',
    thumbnail_details: 'http://.../portrait_incredible.jpg',
    comics: {
      items: []
    },
    series: {
      items: []
    },
    urls: [
      {
        type: 'details',
        url: 'http://...'
      },
    ],
  };

  it('should render correctly', () => {
    const props = {
      item: character,
      onTileDetailsRequested: () => {}
    };

    const wrapper = mount(
      <CharacterTile { ...props } />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should notify about the click on the tile', () => {
    const props = {
      item: character,
      onTileDetailsRequested: jest.fn()
    };

    const component = mount(
      <CharacterTile { ...props } />
    );

    component.find('a').at(0).simulate('click', {});
    expect(props.onTileDetailsRequested).toBeCalledWith(props.item);
  });
});
