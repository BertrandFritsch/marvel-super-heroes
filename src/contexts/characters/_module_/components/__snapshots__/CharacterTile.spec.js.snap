// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterTile should render correctly 1`] = `
<CharacterTile
  item={
    Object {
      "id": "id1",
      "name": "name1",
      "thumbnail": "http://.../standard_xlarge.jpg",
      "thumbnail_details": "http://.../portrait_incredible.jpg",
      "urls": Array [
        Object {
          "type": "details",
          "url": "http://...",
        },
      ],
    }
  }
  onTileDetailsRequested={[Function]}
>
  <div
    className="character-tile"
  >
    <a
      href="#"
      onClick={[Function]}
    >
      <img
        src="http://.../standard_xlarge.jpg"
      />
    </a>
    <span
      className="character-tile-name"
    >
      name1
    </span>
    <div
      className="character-tile-links"
    >
      <a
        href="http://..."
      >
        details
      </a>
    </div>
  </div>
</CharacterTile>
`;
